vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -96,6 +107,29 @@ def check_target_present(cls, target):
             raise ValueError(f"Target {target} not found in model_manager.")
         return target
 
+    def __call__(self, target_to_data_frame: dict[ModelID, pd.DataFrame], current_value: Any):
+        # Only relevant for a dynamic filter.
+        # Although targets are fixed at build time, the validation logic is repeated during runtime, so if a column
+        # is missing then it will raise an error. We could change this if we wanted.
+        targeted_data = self._validate_targeted_data(
+            {target: data_frame for target, data_frame in target_to_data_frame.items() if target in self.targets},
+            eagerly_raise_column_not_found_error=True,
+        )
+
+        if (column_type := self._validate_column_type(targeted_data)) != self._column_type:
+            raise ValueError(
+                f"{self.column} has changed type from {self._column_type} to {column_type}. A filtered column cannot "
+                "change type while the dashboard is running."
+            )
+
+        if isinstance(self.selector, SELECTORS["categorical"]):
+            return self.selector(options=self._get_options(targeted_data, current_value))
```

Comment:
Yes, the `current_value` represent correctly persisted UI value that's propagated into the on-page-load action and then into this (`Filter.__call__()`) function. The current_value is taken into account in a way it's described https://github.com/mckinsey/vizro/pull/879#discussion_r1863284434.

Reply:
And the dash bug means that this doesn't happen for all selectors?

Reply:
So, `current_value` is considered only for calculating new options/min/max (as `current_value` always stays the same for all selectors),  This is done via:
 - `self._get_options(targeted_data, current_value)`,
 -  `self._get_min_max(targeted_data, current_value)`

The comment about the dash bug here is actually related to another line: `self.selector(min=_min, max=_max, current_value=current_value)`. This line calls the `selector.__call__(...)`, and theoretically the `selector.__call__()` should not take the `current_value` into the consideration because only new options/min/max should define the new selector component and the `current_value` should be handled with the `dash persistence` here. 

However, as there is the bug with the dash persistence, for components that use dcc.Store/dbc.Input and clienside_callback to handle syncing multiple UI values, I propagated and used the `current_value` just to hack the solution. This works properly, but it is really messy what you can see from the implementation of the `slider.js`/`range_slider.js` (these two functions will be drastically simplified when the dash bug is fixed).
Also, when dash bug is fixed, we will no longer propagate the `current_value` for the numerical selectors (which is described with the mentioned comment).